src/Microsoft.Dnx.Runtime/ApplicationHostContext.cs

```diff
@@ -206,7 +211,37 @@ private static Result InitializeCore(ApplicationHostContext context)
 
             lockFileLookup?.Clear();
 
-            return new Result(libraries, lockFileExists, validLockFile, lockFileValidationMessage);
+            var lockFilePresent = context.LockFile != null;
+            context.LockFile = null; // LockFile is no longer needed
+
+            return new Result(libraries, lockFilePresent, validLockFile, lockFileValidationMessage);
+        }
+
+        private static LockFileTarget SelectTarget(ApplicationHostContext context, LockFile lockFile)
+        {
+            if (context.RuntimeIdentifiers != null)
```

Comment:
Yes, it can. Sometimes AHC is created with no runtime identifier (such as when being used for compilation, which is runtime-independent)